### PR TITLE
API tests for the Recommendation API

### PR DESCRIPTION
**Pull request by matejnesuta**
## Summary by Sourcery

Add end-to-end API tests for the Recommendation API covering empty input, valid and invalid PURLs, and behavior with duplicates and missing data.

Tests:
- Introduce e2e tests for the Recommendation API responses for empty lists, single and multiple PURLs, and handling of non-existent and duplicate PURLs.
- Add negative tests to ensure invalid PURL formats are rejected with HTTP 400 responses.
- Document a pending test scenario for duplicate vulnerabilities, disabled until the related backend issue is fixed.

### CODE REVIEW

File: e2e/tests/api/features/recommendation.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+//   const recs = res.data.recommendations;
+//   const key = Object.keys(recs)[0];
+//   const item = recs[key][0];
+//   const vulnerabilities = item.vulnerabilities;
+//   const ids = vulnerabilities.map((v: { id: string }) => v.id);
+//   const uniqueIds = new Set(ids);
+
+//   expect(uniqueIds.size).toBe(ids.length);
+// });
```
[the remark this answers is not in the document]
Comment (matejnesuta) [reply e2e/tests/api/features/recommendation.ts]: Looks like the bug is already fixed, so I have removed the comment.
